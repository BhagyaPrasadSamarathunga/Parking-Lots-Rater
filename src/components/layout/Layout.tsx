import { ReactNode } from "react";
import Header from "../header/Header";

interface LayoutProps {
  children?: ReactNode;
}

const Layout = (props: LayoutProps): JSX.Element => {
  const {children} = props;

  return (
    <div className="flex flex-col h-screen w-full bg-gray-300 overflow-hidden" >
       <Header />
      <div className="flex flex-col flex-grow h-full min-h-content bg-whitemt-[52px] lg:mt-[60px] overflow-hidden">
        {children}
      </div>
    </div>
  )
}

export default Layout;