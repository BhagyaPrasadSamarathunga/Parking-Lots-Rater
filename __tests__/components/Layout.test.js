import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import Layout from '../../src/components/layout/Layout';

describe('Layout component', () => {

  test('matches snapshot', () => {
    const { asFragment } = render(<Layout><div>Home</div></Layout>);
    expect(asFragment()).toMatchSnapshot();
  });

  test('renders children components', () => {
    render(<Layout><div>Home</div></Layout>);
    const layoutElement = screen.getByText(/Home/i);
    expect(layoutElement).toBeInTheDocument();
  });

  test('renders header component with the given content', () => {
    render(<Layout><div>Home</div></Layout>);
    const headerElement = screen.getByText(/Park Rater/i);
    expect(headerElement).toBeInTheDocument();
  });
});