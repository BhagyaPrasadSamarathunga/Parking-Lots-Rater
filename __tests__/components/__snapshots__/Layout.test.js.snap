// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="flex flex-col h-screen w-full bg-gray-300 overflow-hidden"
  >
    <div
      class="flex justify-between items-center h-[52px] lg:h-[60px] z-20 fixed w-full top-0 left-0 bg-sky-600"
    >
      <p
        class="text-white font-semibold ml-6 text-lg"
      >
        Park Rater
      </p>
    </div>
    <div
      class="flex flex-col flex-grow h-full min-h-content bg-whitemt-[52px] lg:mt-[60px] overflow-hidden"
    >
      <div>
        Home
      </div>
    </div>
  </div>
</DocumentFragment>
`;
